napari/conftest.py

```diff
@@ -449,7 +449,8 @@ def _mock_app():
     `app_model.Application` in the test. It does not however, register Qt
     related actions or providers. If this is required for a unit test,
     `napari._qt._qapp_model.qactions.init_qactions()` can be used within
-    the test.
+    the test. Please note as `init_qactions()` is `lru_cache` decorated you
+    may need to call `init_qactions.cache_clear()` first.
```

Comment:
this is autouse mock. So these mocs should call clean cache on their own responsibility.

Reply:
We decided when mocking `app` to register app-model actions but NOT q actions as not all tests will necessarily want q actions registered - thus we suggest if a test wants q actions registered for them to add `init_qactions()` to the test.

Thus we have have not cleared the cache automatically, since we do not automatically register q actions. Happy to hear your thoughts though?

Reply:
my main wory is that this is an automatically applied fixture so less knowledgeable database person may do not reach this docstring.

Reply:
Yes I understand the concern. What do you suggest we do?

Reply:
Yeah we didn't love this from the beginning. I'll also note that this seems to be affecting the `make_napari_viewer` fixture, since the cache seems to outlast a single test execution. E.g. in `test_dock_widget_toggler` up there we have to explicitly register the `window` provider, even though `make_napari_viewer` should be calling `init_qactions`. `make_napari_viewer` should *always* leave the `viewer` in a complete state imo, so we may need to clear cache all the time, even if we don't call `init_qactions`?